code/game/machinery/bank_machine.dm

```diff
@@ -79,21 +80,30 @@
 
 	return data
 
-/obj/machinery/computer/bank_machine/ui_act(action, params)
+/obj/machinery/computer/bank_machine/ui_act(action, params, datum/tgui/ui)
 	if(..())
 		return
 
 	switch(action)
 		if("siphon")
 			say("Siphon of station credits has begun!")
-			siphoning = TRUE
+			start_siphon(ui.user)
```

Comment:
```suggestion
			if(iscarbon(usr))
				var/mob/living/carbon/C = usr
				start_siphon(C)
```

Reply:
Other code uses this, likely a better choice as I see no example of yours besides in slime people